libcontainer/cgroups/fs/rdma.go

```diff
@@ -0,0 +1,118 @@
+// +build linux
+
+package fs
+
+import (
+	"github.com/opencontainers/runc/libcontainer/cgroups"
+	"github.com/opencontainers/runc/libcontainer/cgroups/fscommon"
+	"github.com/opencontainers/runc/libcontainer/configs"
+	"math"
+	"strconv"
+	"strings"
+)
+
+type RdmaGroup struct {
+}
+
+func (s *RdmaGroup) Name() string {
+	return "rdma"
+}
+
+func (s *RdmaGroup) Apply(path string, d *cgroupData) error {
+	return join(path, d.pid)
+}
+
+func createCmdString(device string, limits configs.LinuxRdma) string {
+	var cmdString string
+	cmdString = device
+	if limits.HcaHandles != nil {
+		cmdString = cmdString + " " + "hca_handle=" + strconv.FormatUint(uint64(*limits.HcaHandles), 10)
+	}
+	if limits.HcaObjects != nil {
+		cmdString = cmdString + " " + "hca_object=" + strconv.FormatUint(uint64(*limits.HcaObjects), 10)
+	}
+	return cmdString
+}
+
+func (s *RdmaGroup) Set(path string, cgroup *configs.Cgroup) error {
+	for device, limits := range cgroup.Resources.Rdma {
+		if err := fscommon.WriteFile(path, "rdma.max", createCmdString(device, limits)); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func (s *RdmaGroup) GetStats(path string, stats *cgroups.Stats) error {
+	if !cgroups.PathExists(path) {
+		return nil
+	}
+	currentData, err := fscommon.GetCgroupParamString(path, "rdma.current")
+	if err != nil {
+		return err
+	}
+	currentPerDevices := strings.Split(string(currentData), "\n")
```

Comment:
1. No need to convert `currentData` to `string` type -- it is already a string.
2. You'd better use `fscommon.ReadFile` here -- `GetCgroupParamString` is tailored more for single-value files.

Reply:
Fixed.